packages/medusa/src/api-v2/store/customers/route.ts

```diff
@@ -1,14 +1,37 @@
 import { MedusaRequest, MedusaResponse } from "../../../types/routing"
-import { createCustomerAccountWorkflow } from "@medusajs/core-flows"
+
+import { ContainerRegistrationKeys } from "@medusajs/utils"
 import { CreateCustomerDTO } from "@medusajs/types"
+import { createCustomerAccountWorkflow } from "@medusajs/core-flows"
 
 export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  if (req.auth_user?.app_metadata?.customer_id) {
+    const remoteQuery = req.scope.resolve(
+      ContainerRegistrationKeys.REMOTE_QUERY
+    )
+    const query = {
+      customer: {
+        __args: { id: req.session.auth_user.app_metadata.customer_id },
```

Comment:
```suggestion
        __args: { id: req.auth_user?.app_metadata?.customer_id },
```
since you checked for existence for this already

Reply:
are the optional chaining necessary since the i already check its existence? I know in the latest TS version they have improved this